lib/conjur/puppet_module/config.rb

```diff
@@ -14,24 +14,24 @@ def load
         end
 
         def from_file
-          if File.file?(CONFIG_FILE_PATH)
-            c = YAML.safe_load(File.read(CONFIG_FILE_PATH))
+          return {} unless File.file?(CONFIG_FILE_PATH)
 
-            if c['cert_file']
-              raise "Cert file '#{c['cert_file']}' cannot be found!" unless File.file?(c['cert_file'])
+          c = YAML.safe_load(File.read(CONFIG_FILE_PATH))
 
-              c['ssl_certificate'] ||= File.read c['cert_file']
-            end
+          if c['cert_file']
+            raise "Cert file '#{c['cert_file']}' cannot be found!" unless File.file?(c['cert_file'])
 
-            c
-          else
-            {}
+            c['ssl_certificate'] ||= File.read c['cert_file']
           end
+
+          c
         end
 
         # We do this in a method to allow for easier testing
         def load_registry_module
+          # :nocov:
```

Comment:
What is this? I'm hoping that might also answer why it's needed on both sides of the require.

Reply:
It's to ignore it from code coverage metrics. This line isn't needed in tests since we mock the class invocation but code coverage then marks it as unvisited. The flag `:nocov:` works in blocks to surround the code to willingly exclude it from simplecov results so we add it it at the start and end of the code we are ignoring. https://github.com/simplecov-ruby/simplecov#ignoringskipping-code